modules/operations/cycle_highway_tag.js

```diff
@@ -12,7 +12,8 @@ export function operationCycleHighwayTag(context, selectedIDs) {
   // Allow cycling through lines that match these presets
   const allowPresetRegex = [
     /^highway\/(motorway|trunk|primary|secondary|tertiary|unclassified|residential|living_street|service|track)/,
-    /^line$/
```

Comment:
This isn't correct- this code has modified the existing highway tag change functionality and also allowed it to work on crossings, and it has added a bunch of crossing ways to the tags we're cycling through. 

We need two separate preset regexes, and two separate preset ID lists, one for highways and one for crossings. 

The code here has combined the two, so that the user may cycle a tag on either a highway or a crossing, and they could cycle seamlessly from highway types into crossing types- this is not what we want. 

We'd need to reafctor a bit- instead of 1 `allowPresetRegex`, rename it to `allowHighwayPresetRegex` and then create another similar one, `allowCrossingPresetRegex`, etc.